tagstudio/src/qt/widgets/preview_panel.py

```diff
@@ -601,28 +712,15 @@ def update_widgets(self) -> bool:
                 # TODO: Do this all somewhere else, this is just here temporarily.
                 ext: str = filepath.suffix.lower()
                 try:
-                    if filepath.suffix.lower() in [".gif"]:
-                        with open(filepath, mode="rb") as file:
-                            if self.preview_gif.movie():
-                                self.preview_gif.movie().stop()
-                                self.gif_buffer.close()
-
-                            ba = file.read()
-                            self.gif_buffer.setData(ba)
-                            movie = QMovie(self.gif_buffer, QByteArray())
-                            self.preview_gif.setMovie(movie)
-                            movie.start()
-
-                        image = Image.open(str(filepath))
-                        self.resizeEvent(
-                            QResizeEvent(
-                                QSize(image.width, image.height),
-                                QSize(image.width, image.height),
-                            )
-                        )
-                        self.preview_img.hide()
-                        self.preview_vid.hide()
-                        self.preview_gif.show()
+                    if MediaCategories.is_ext_in_category(
+                        ext, MediaCategories.IMAGE_ANIMATED_TYPES, mime_fallback=True
+                    ):
+                        with open(filepath, mode="rb") as f:
+                            file_bytes: bytes = f.read()
+                            image: Image.Image = Image.open(io.BytesIO(file_bytes))
+                            if hasattr(image, "n_frames"):
+                                if image.n_frames > 1:
+                                    self.set_anim_img(filepath, ext, file_bytes, image)
```

Comment:
Just an approval comment: I like the check to see if an image has multiple frames - for some reason Qt displays QMovies a bit strangely compared to static images, so this check is nice for improving the quality of non-animated images that share extensions with animated counterparts (i.e. gif).